esp-hal/src/gpio/interconnect.rs

```diff
@@ -1,11 +1,102 @@
-//! Peripheral signal interconnect using IOMUX or GPIOMUX.
-
-// FIXME: https://github.com/esp-rs/esp-hal/issues/2954 The GPIO implementation does not contain any
-// locking. This is okay there, because the implementation uses either W1TS/W1TC
-// registers to set certain bits, or separate registers for each pin - and there
-// can only be one instance of every GPIO struct in safe code. However, with the
-// interconnect module we allow multiple handles, which means possible RMW
-// operations on the pin registers cause data races.
+//! # Peripheral signal interconnect using the GPIO matrix.
+//!
+//! The GPIO matrix offers flexible connection options between GPIO pins and
+//! peripherals. This module offers capabilities not covered by GPIO pin types
+//! and drivers, like routing fixed logic levels to peripheral inputs, or
+//! inverting input and output signals.
```

Comment:
Great docs!

nitpick again: there is no space between the text and the link, can we link to a relevant chapter (e.g. "iomuxgpio"?)